File: algorithms/Codility-master/Codility-master/3_TimeComplexity/PermMissingElem.py

# 30%
def solutionC(A):
    n = len(A) + 1
    nums = (int((n+1)/2))*n
    
    return nums - sum(A)

# 100%
def solution(A):
    if len(A) == 0:
        return 1
    if (len(A) == 1) and (A[0] != 1):
        return 1
        
    n = len(A) + 1
    nums = int(((n+1)/2)*n)
    for num in A:
        nums -= num
    return nums

File: algorithms/Codility-master/Codility-master/3_TimeComplexity/PermMissingElem.js


// 100%
function solution(A) {
    // write your code in JavaScript (Node.js 8.9.4)
    if (A.length == 0){
        return 1;
    }
    if ((A.length == 1) && (A[0] != 1)){
        return 1;
    }
    
    let n = A.length + 1;
    var nums = Math.floor(((n+1)/2)*n);
    
    for(let i=0; i<A.length ; i++){
        nums -= A[i];
    }
        
    return nums;
}